### PR TITLE
quotactl() API

### PR DESCRIPTION
**Pull request by polachok**
I'm fairly new to rust, so I'm looking for some early feedback on this API. Am I moving into right direction with this?

### CODE REVIEW

File: src/sys/quotactl.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+use {Result, NixPath, from_ffi};
+use errno::Errno;
+use libc::{c_int, c_char};
+
+#[cfg(all(target_os = "linux",
+          any(target_arch = "x86",
+              target_arch = "x86_64",
+              target_arch = "arm")),
+          )]
+pub mod quotactl {
```
Comment (carllerche) [inline src/sys/quotactl.rs]: I am not sure why this module needs to be public.

**Comment by carllerche** [comment]
Looks mostly fine to me. I added a minor nitpick. API style wise, nix doesn't really have a consistent guideline yet (something I am hoping to work on), what you came up with is in line with other APIs in nix though... so seems good :)

File: src/sys/mod.rs [hunk 2]
```diff
@@ -47,3 +47,10 @@ pub mod time;
               target_arch = "arm")),
           )]
 pub mod ptrace;
+
+#[cfg(all(target_os = "linux",
+          any(target_arch = "x86",
+              target_arch = "x86_64",
+              target_arch = "arm")),
+          )]
+pub mod quotactl;
```
Comment (utkarshkukreti) [inline src/sys/mod.rs]: This should probably be named `quota` as the corresponding C header file is `sys/quota.h`.

Comment (polachok) [inline src/sys/mod.rs]: fixed

File: src/sys/quota.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+pub fn quotactl_get<P: ?Sized + NixPath>(which: quota::QuotaType, special: &P, id: c_int, dqblk: &mut quota::Dqblk) -> Result<()> {
+	use std::mem;
+	unsafe {
+		quotactl(quota::QuotaCmd::new(quota::Q_GETQUOTA, which), Some(special), id, mem::transmute(dqblk))
+	}
+}
+
+pub fn quotactl_set<P: ?Sized + NixPath>(which: quota::QuotaType, special: &P, id: c_int, dqblk: &mut quota::Dqblk) -> Result<()> {
```
Comment (polachok) [inline src/sys/quota.rs]: quotactl() is not supposed to mutate dqblk here, would it be safe to pass a &quota::Dqblk and then transmute it?

**Comment by polachok** [comment]
I think it's ready, what do you think?

**Comment by carllerche** [comment]
Merged! Thanks (8da9b314e973e756c8384a6110fb85c899d64561)

**Pull request closed without merging**